Windows Cross-Build fails: unguarded std::os::unix::fs::PermissionsExt in make_tree_owner_traversable

Windows Cross-Build is failing on main (`d7656e3`):

```
error[E0433]: cannot find `unix` in `os`
error: could not compile `artifact-keeper-backend` (lib)
```

Source: `backend/src/services/incus_scanner.rs:470` — `make_tree_owner_traversable` uses `std::os::unix::fs::PermissionsExt` without `cfg(unix)` gating. The function was added by the recent #1492 work and works on Linux, but the lib target can no longer compile on Windows.

Fix: gate the function on `#[cfg(unix)]` with a `#[cfg(not(unix))]` no-op stub. Incus is a Linux concept, so a Windows no-op is acceptable — the call site at line 453 still works on both platforms.

Blocking v1.2.0-rc.3.

--- a/backend/src/services/incus_scanner.rs
+++ b/backend/src/services/incus_scanner.rs
@@ -466,6 +466,14 @@ impl IncusScanner {
     /// in-tree `var/run -> /run` (or any absolute link) cannot cause us to
     /// chmod a path outside the workspace — the property `chmod -R` cannot
     /// guarantee. Only owner bits are added; group/other are never widened.
+    ///
+    /// Unix-only: Windows has no POSIX permission bits and incus images are a
+    /// Linux concept, so the Windows lib build gets a no-op stub below (#1525)
+    /// to keep the crate cross-compiling cleanly for the Windows CLI target.
+    /// The Windows scanner code path is never reached in practice; the stub
+    /// just lets `extract_tarball_to_dir` stay platform-agnostic at the call
+    /// site.
+    #[cfg(unix)]
     fn make_tree_owner_traversable(root: &Path) -> Result<()> {
         use std::os::unix::fs::PermissionsExt;
 
@@ -520,6 +528,16 @@ impl IncusScanner {
                 }
             }
         }
+        Ok(())
+    }
+
+    /// Windows no-op stub for `make_tree_owner_traversable` (see #1525). Windows
+    /// has no POSIX mode bits, the incus scanner is never invoked on Windows in
+    /// practice, and the lib must still cross-compile cleanly for the Windows
+    /// CLI build. Returns `Ok(())` so the platform-agnostic call site at
+    /// `extract_tarball_to_dir` continues to work.
+    #[cfg(not(unix))]
+    fn make_tree_owner_traversable(_root: &Path) -> Result<()> {
         Ok(())
     }
 

